### ISSUE TITLE
[xcode11] watchOS simulator projects fail to build

### ISSUE THREAD

**Issue by rolfbjarne** [xamarin/xamarin-macios#6252]
Building for watchOS simulator (and probably device as well) fails with:

```
clang : error : linker command failed with exit code 1 (use -v to see invocation) [/Users/builder/jenkins/workspace/xamarin-macios-pr-builder/tests/xharness/tmp-test-dir/dont link-watchos-extension/dont link-watchos-extension.csproj] 
MTOUCH : error MT5210: Native linking failed, undefined symbol: _main. Please verify that all the necessary frameworks have been referenced and native libraries are properly linked in. [/Users/builder/jenkins/workspace/xamarin-macios-pr-builder/tests/xharness/tmp-test-dir/dont link-watchos-extension/dont link-watchos-extension.csproj] 
MTOUCH : error MT5202: Native linking failed. Please review the build log. [/Users/builder/jenkins/workspace/xamarin-macios-pr-builder/tests/xharness/tmp-test-dir/dont link-watchos-extension/dont link-watchos-extension.csproj] 
```

https://jenkins.mono-project.com/job/xamarin-macios-pr-builder/10186/Test_20Report/.

**Comment by spouliot** [comment]
Most failures seems fixed but I expected this one to be too
```
Xamarin.iOS.Tasks.WatchKit2("iPhoneSimulator").BasicTest: #RunTarget-ErrorCount
    redefinition of 'main'
    Failed to compile the file(s) '/Users/builder/jenkins/workspace/xamarin-macios-pr-builder/msbuild/tests/MyWatchKit2IntentsExtension/obj/iPhoneSimulator/Debug/mtouch-cache/i386/main.m'. Please file a bug report at https://github.com/xamarin/xamarin-macios/issues/new
Expected: 0
But was: 2
```
https://jenkins.mono-project.com/job/xamarin-macios-pr-builder/10296//Test_20Report

**Comment by rolfbjarne** [comment]
@spouliot 

> Most failures seems fixed but I expected this one to be too
> 
> ```
> Xamarin.iOS.Tasks.WatchKit2("iPhoneSimulator").BasicTest: #RunTarget-ErrorCount
>     redefinition of 'main'
>     Failed to compile the file(s) '/Users/builder/jenkins/workspace/xamarin-macios-pr-builder/msbuild/tests/MyWatchKit2IntentsExtension/obj/iPhoneSimulator/Debug/mtouch-cache/i386/main.m'. Please file a bug report at https://github.com/xamarin/xamarin-macios/issues/new
> Expected: 0
> But was: 2
> ```
> 
> https://jenkins.mono-project.com/job/xamarin-macios-pr-builder/10296//Test_20Report

fixed in #6323.

**Comment by rolfbjarne** [comment]
This issue was fixed in #6294.
